src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,221 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarByIdStatement =
+                        connection.prepareStatement(gerCarByIdRequest)) {
+            getCarByIdStatement.setLong(1, id);
+            ResultSet resultSet = getCarByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getAllDriversByCar(car));
+            return Optional.of(car);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllRequest = "SELECT c.id as car_id, c.name as car_name,"
+                + " m.id as manufacturer_id, m.name as manufacturer_name,"
+                + " m.country as manufacturer_country"
+                + " FROM cars c JOIN manufacturers m on m.id = c.manufacturer_id"
+                + " WHERE c.is_deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAllRequest)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant get all cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getAllDriversByCar(car));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET name = ?, manufacturer_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getName());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant update car " + car, e);
+        }
+        deleteAllDriversByCar(car);
+        addDriversToCar(car, car.getDrivers());
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteByIdRequest = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteByIdStatement =
+                        connection.prepareStatement(deleteByIdRequest)) {
+            deleteByIdStatement.setLong(1, id);
+            int updatedRows = deleteByIdStatement.executeUpdate();
+            return updatedRows >= 0;
```

Comment:
Here you return true even number of deleted cars is 0. do you think this behavior is OK?